cmd/app-server/linterapp/linterapp.go

```diff
@@ -0,0 +1,109 @@
+package linterapp
+
+import (
+        "context"
+        "fmt"
+        "log"
+        "net/http"
+        "os"
+        "io/ioutil"
+        "strings"
+        "encoding/json"
+        "github.com/apigee/registry/cmd/registry/cmd"
+        "github.com/apigee/registry/rpc"
+	    "github.com/apigee/registry/connection"
+        "cloud.google.com/go/compute/metadata"
+	    "github.com/golang/protobuf/jsonpb"
+)
+
+type PubSubMessage struct {
+        Message struct {
+                Data []byte `json:"data,omitempty"`
+                ID   string `json:"id"`
+        } `json:"message"`
+        Subscription string `json:"subscription"`
+}
+
+func getAuthToken() (string, error) {
+        serviceURL := "https://" + strings.Split(os.Getenv("APG_REGISTRY_ADDRESS"), ":")[0]
+        tokenURL := fmt.Sprintf("/instance/service-accounts/default/identity?audience=%s", serviceURL)
+        idToken, err := metadata.Get(tokenURL)
+        if err != nil {
+                log.Printf("metadata.Get: failed to query id_token: %+v", err)
+                return "", err
+        }
+
+        return idToken, nil
+}
+
+func readPubsubMessage(w http.ResponseWriter, r *http.Request) string {
+        var m PubSubMessage
+        body, err := ioutil.ReadAll(r.Body)
+        if err != nil {
+                log.Printf("ioutil.ReadAll: %v", err)
+                http.Error(w, "Bad Request", http.StatusBadRequest)
+                return ""
+        }
+        if err := json.Unmarshal(body, &m); err != nil {
+                log.Printf("json.Unmarshal: %v", err)
+                http.Error(w, "Bad Request", http.StatusBadRequest)
+                return ""
+        }
+
+        data := string(m.Message.Data)
+        log.Printf("%s", data)
+        return data
+}
+
+func RequestHandler(w http.ResponseWriter, r *http.Request) {
+
+    data := readPubsubMessage(w, r)
+    if data == "" {
+        return
+    }
+    message := rpc.Notification{}
+    if err := jsonpb.UnmarshalString(data, &message); err != nil {
+        log.Printf("json.Unmarshal: %v", err)
+        fmt.Fprintf(w, "Wrong message format")
+        return
+    }
+
+    if message.Change != rpc.Notification_DELETED {
```

Comment:
Should this run for any notification types added in the future? If not, consider checking against specific notifications (e.g. CREATED, UPDATED).

It's also a common pattern ([example](https://golang.org/doc/effective_go.html#if)) to exit early on error/exit conditions rather than wrapping the regular flow in an if-statement. Something like:
```
switch type := message.Change {
    rpc.Notification_CREATED, rpc.Notification_UPDATED:
        log.Printf("running linter for change type %q", type)
    default:
        log.Printf("ignoring change type %q for linting", type)
        return
}
// code for processing CREATED, UPDATED types
```

Reply:
Thanks for this suggestion, much better approach. Done :)